reconcile/utils/oc.py

```diff
@@ -319,6 +364,78 @@ def __init__(
             "LOG_SLOW_OC_RECONCILE", ""
         ).lower() in ["true", "yes"]
 
+    def _init(
+        self,
+        connection_parameters: OCConnectionParameters,
+        init_projects: bool = False,
+        init_api_resources: bool = False,
+        local: bool = False,
+    ):
+        self.cluster_name = connection_parameters.cluster_name
+        self.server = connection_parameters.server_url
+        oc_base_cmd = ["oc", "--kubeconfig", "/dev/null"]
+        if connection_parameters.skip_tls_verify:
+            oc_base_cmd.extend(["--insecure-skip-tls-verify"])
+        if self.server:
+            oc_base_cmd.extend(["--server", self.server])
+
+        token = connection_parameters.automation_token
+        if (
+            connection_parameters.is_cluster_admin
+            and connection_parameters.cluster_admin_automation_token
+        ):
+            token = connection_parameters.cluster_admin_automation_token
```

Comment:
Can use conditional assign to avoid mutation
```python
        token = connection_parameters.cluster_admin_automation_token if (
                connection_parameters.is_cluster_admin
                and connection_parameters.cluster_admin_automation_token
        ) else connection_parameters.automation_token
```

Reply:
To be honest I'm not a big fan of multi-line conditional assignments. I find them harder to read than simple old-style if assignments, as they are common in other languages too - but yeah that is a taste question I guess :D
I could reduce the length of the parameter names - which would require additional tmp assignments again.

I guess the best would be like you suggest below to extract this into another method. Will keep it for now and address this in the follow-up PRs, when it is more clear to me when `automation_token` is used over `admin_token`.

Reply:
To simplify code for this instance, extract function is preferred over extract variable, as you mentioned tmp assign is not clean. Is it a real case that `is_cluster_admin` is `True` but `cluster_admin_automation_token` is `None`? If that case exists, then `automation_token` will be used even `is_cluster_admin`, which is different than the other logic in `OCNative`.

 `token` is only used from line 382 to line 390, can use assignment expression in `if` to make it clear, like this:


```python
def get_token(connection_parameters):
    if (
            connection_parameters.is_cluster_admin
            and connection_parameters.cluster_admin_automation_token
    ):
        return connection_parameters.cluster_admin_automation_token
    return connection_parameters.automation_token
```

```python
if token := get_token(connection_parameters):
    oc_base_cmd.extend(["--token", token])
```

if that null check for `cluster_admin_automation_token` is not needed, then it's even cleaner.